src/Shared/Recruit.Vacancies.Client/Infrastructure/Reports/CsvBuilder.cs

```diff
@@ -43,49 +48,42 @@ private static CsvWriter GetCsvWriter(TextWriter streamWriter)
             return csv;
         }
 
-        private void WriteNoResults(CsvWriter csv, DateTime reportDate)
+        private void WriteNoResults(CsvWriter csv, IEnumerable<KeyValuePair<string, string>> headers)
         {
-            WriteTotalHeader(csv, reportDate, 0, 2);
+            WriteTotalHeader(csv, headers);
         }
 
-        private void WriteTotalHeader(CsvWriter csv, DateTime reportDate, int total, int fieldCount)
+        private void WriteTotalHeader(
+            CsvWriter csv,
+            IEnumerable<KeyValuePair<string, string>> headers)
         {
+            headers = headers ?? new KeyValuePair<string, string>[0];
             csv.WriteField("PROTECT");
 
-            for (var i = 1; i < fieldCount; i++)
-            {
-                csv.WriteField("");
-            }
-            csv.NextRecord();
-
-            WriteEmptyRow(csv, fieldCount);
-            WriteEmptyRow(csv, fieldCount);
-
-            csv.WriteField("Date");
-            csv.WriteField("Total_Number_Of_Applications");
+            WriteEmptyRow(csv, 0);
+            WriteEmptyRow(csv, 0);
+            WriteEmptyRow(csv, 0);
 
-            for (var i = 2; i < fieldCount; i++)
+            if (headers.Any())
             {
-                csv.WriteField("");
-            }
-            csv.NextRecord();
-
-            csv.WriteField(reportDate.ToString(DateTimeFormat));
-            csv.WriteField(total);
+                foreach (var kvp in headers)
```

Comment:
I think you need to pad out the rows with empty fields so each row has the same number of columns